doc/docs/Python_Tutorials/Adjoint_Solver.md

```diff
@@ -2,13 +2,714 @@
+The adjoint solver supports inverse design and [topology
+optimization](https://en.wikipedia.org/wiki/Topology_optimization) by
+providing the functionality to wrap an optimization library around the
+gradient computation. This is demonstrated in several tutorials below.
+
+[TOC]
+
+Broadband Waveguide Mode Converter with Minimum Feature Size
+------------------------------------------------------------
+
+This example demonstrates some of the advanced functionality of the
+adjoint solver including worst-case (minimax) optimization, multiple
+objective functions, and constraints on the minimum line width and
+line spacing. The design problem involves a broadband waveguide mode
+converter with minimum feature size in 2D. This is based on
+[M.F. Schubert et al., ACS Photonics, Vol. 9, pp. 2327-36,
+(2022)](https://doi.org/10.1021/acsphotonics.2c00313).
+
+The mode converter must satisfy two separate design objectives: (1)
+minimize the reflectance $R$ into the fundamental mode of the input
+port ($\mathcal{S}_{11}$) and (2) maximize the transmittance $T$ into
+the second-order mode of the output port ($\mathcal{S}_{21}$). There
+are different ways to define this multi-objective and multi-wavelength
+optimization. The approach taken here is *worst-case* optimization
+whereby we minimize the maximum (the worst case) of $R$ and $1-T$
+across all six wavelengths (in the O-band for
+telecommunications). This is known as *minimax* optimization.
+
+The challenge with minimax optimization is that the $\max$ objective
+function is not everywhere differentiable. This property would seem to
+preclude the use of gradient-based optimization algorithms for this
+problem which involves 12 independent functions ($R$ and $1-T$ for
+each of six wavelengths). Fortunately, there is a workaround: the
+problem can be reformulated as a differentiable problem using
+a so-called "epigraph" formulation: introducing
+a new "dummy" optimization variable $t$ and
+adding each independent function $f_k(x)$ as a new nonlinear constraint $t \ge f_k(x)$. See
+the [NLopt
+documentation](https://nlopt.readthedocs.io/en/latest/NLopt_Introduction/#equivalent-formulations-of-optimization-problems)
+for an overview of this approach. The minimax/epigraph approach is
+also covered in the [near-to-far field
+tutorial](https://nbviewer.org/github/NanoComp/meep/blob/master/python/examples/adjoint_optimization/06-Near2Far-Epigraph.ipynb).
+
+In this example, we use a minimum feature size of 150 nm for the line
+width *and* line spacing. The implementation of these constraints is
+based on [A.M. Hammond et al., Optics Express, Vol. 29, pp. 23916-38
+(2021)](https://doi.org/10.1364/OE.431188).
+
+There are five important items to note in the set up of the
+optimization problem:
+
+- The lengthscale constraint is activated only in the final epoch. The
+  initial design of the final epoch should be binary. This is because
```

Comment:
```suggestion
- The lengthscale constraint is activated only in the final epoch. It is
   often helpful to mostly binarize the design before this final epoch. This is because
```